src/ccache.cpp

```diff
@@ -228,7 +228,9 @@ guess_compiler(string_view path)
 #endif
 
   const string_view name = Util::base_name(compiler_path);
-  if (name.find("clang") != nonstd::string_view::npos) {
+  if (name.find("clang-cl") != nonstd::string_view::npos) {
+    return CompilerType::clangcl;
```

Comment:
This is starting to get a bit complicated. I don't want to end up in a situation where there is a plethora of compiler types, each describing details of compilers. That has to be handled by other means. See also #958 for what I envision will be in future change in how compiler detection works.

What would happen if clang-cl was just characterized as cl instead? (On a side note, a better name would be "msvc" which I plan to rename "cl" to. Compiler type is _type_, not necessarily the actual compiler name.) Put differently, is there a case where clang-cl behaves different enough from cl that we need to handle it specially?

Reply:
clang-cl supports also (some of) clang's flags like `-W*` and many `-f*`. See https://clang.llvm.org/docs/UsersManual.html#id10

Reply:
Right, but that doesn't mean that we have to explicitly _not_ handle `-W*` and `-f*` for cl/MSVC. For example, there is no logic for handling `-fdirectives-only` only if we know that we're dealing with GCC, or handle `-F` only for cl/MSVC. Compiler type only needs to be considered when ccache has to treat an option differently for different compilers. I would like to avoid adding more complexity in this area (see also #956).

So I think my question _"What would happen if clang-cl was just characterized as cl instead?"_ is still relevant.

Reply:
@llunak Can you justify it?

Reply:
Done